src/handler/guild/role.ts

```diff
@@ -0,0 +1,266 @@
+                subcommand
+                    .setName("info")
+                    .setDescription("List configured roles")
+            )
+            .toJSON();
+    }
+
+    @Command({ name: "role" })
+    async role(interaction: CommandInteraction): Promise<void> {
+        const subCommand = interaction.options.getSubcommand(true);
+
+        switch (subCommand) {
+            case "add":
+                await this.roleAdd(interaction);
+                break;
+
+            case "remove":
+                await this.roleRemove(interaction);
+                break;
+            default:
+                throw new BotError("Not implemented");
+        }
+    }
+
+    async roleAdd(interaction: CommandInteraction): Promise<void> {
+        await interaction.deferReply();
+
+        const role = interaction.options.getRole("role", true);
+
+        switch (interaction.options.getString("type")) {
+            case "admin": {
+                const commandRepository =
+                    getCustomRepository(CommandRepository);
+                const commandIds =
+                    await commandRepository.getGuildCommandIdsByName(
+                        Role.MODERATOR_COMMANDS,
+                        interaction.guildId
+                    );
+                if (!commandIds || !interaction.guild)
+                    throw new BotError("Something went wrong");
+
+                await mergePermissions(
+                    interaction.guild,
+                    commandIds,
+                    role.id,
+                    "ROLE"
+                );
+
+                const permissionRepository =
+                    getCustomRepository(PermissionRepository);
+
+                // repository upsert is broken
+                await permissionRepository
+                    .createQueryBuilder()
+                    .insert()
+                    .into(Permission)
+                    .values(
+                        commandIds.map((command_id) => ({
+                            id: role.id,
+                            type: PermissionType.ROLE,
+                            command: { command_id },
+                        }))
+                    )
+                    .orUpdate(["id"], ["id", "command_id"])
+                    .execute();
+
+                const success = new MessageEmbed()
+                    .setDescription(
+                        `Successfully added admin role <@&${role.id}>!`
+                    )
+                    .setColor("DARK_GREEN");
+                await interaction.editReply({ embeds: [success] });
+                return;
+            }
+            default:
+                throw new BotError("Not implemented");
+        }
+    }
+
+    async roleRemove(interaction: CommandInteraction): Promise<void> {
+        await interaction.deferReply();
+
+        const role = interaction.options.getRole("role");
+
+        switch (interaction.options.getString("type")) {
+            case "admin": {
```

Comment:
not sure this an appropriate use of a switch statement. since there is only one case.

Reply:
That's because the other cases aren't implemented yet.

I was going to implement them all in this PR, but I underestimated how complicated that would be and wanted to merge this first before I introduce even more changes